WordPress/Helpers/RulesetPropertyHelper.php

```diff
@@ -43,15 +43,24 @@ final class RulesetPropertyHelper {
 	 * @since 2.0.0  No longer supports custom array properties which were incorrectly
 	 *               passed as a string.
 	 * @since 3.0.0  Moved from the Sniff class to this class.
+	 * @since 3.1.0  Added a new parameter to lowercase array keys and values.
 	 *
-	 * @param array $custom Custom list as provided via a ruleset.
-	 * @param array $base   Optional. Base list. Defaults to an empty array.
-	 *                      Expects `value => true` format when `$flip` is true.
-	 * @param bool  $flip   Optional. Whether or not to flip the custom list.
-	 *                      Defaults to true.
+	 * @param array $custom             Custom list as provided via a ruleset.
+	 * @param array $base               Optional. Base list. Defaults to an empty array.
+	 *                                  Expects `value => true` format when `$flip` is true.
+	 * @param bool  $flip               Optional. Whether or not to flip the custom list.
+	 * @param bool  $lowercaseKeyValues Optional. Whether to lowercase keys and values in the resulting array.
+	 *                                  Defaults to false.
 	 * @return array
 	 */
-	public static function merge_custom_array( $custom, array $base = array(), $flip = true ) {
+	public static function merge_custom_array( $custom, array $base = array(), $flip = true, $lowercaseKeyValues = false ) {
+		if ( $lowercaseKeyValues ) {
+			$base   = array_map( 'strtolower', $base );
+			$custom = array_map( 'strtolower', $custom );
+			$base   = array_change_key_case( $base );
+			$custom = array_change_key_case( $custom );
+		}
```

Comment:
Aside from the fact that the values shouldn't be touched, the `$custom` array should not be changed yet at this point, only the `$base` (with `array_change_key_case()`).

The `array_change_key_case()` for `$custom` should be done after the potential flip and before the check for empty `$base`.

Also never a bad thing to explicitly pass the `CASE_LOWER` constant as the second parameter to make the code more readable for the next person (prevents them having to look up what the default is).